common/FileSystem.cpp

```diff
@@ -284,6 +284,107 @@ bool Path::IsAbsolute(const std::string_view path)
 #endif
 }
 
+bool Path::IsCharacterSafeForShell(char c)
+{
+	if (c >= '0' && c <= '9')
+		return true;
+	if (c >= 'a' && c <= 'z')
+		return true;
+	if (c >= 'A' && c <= 'Z')
+		return true;
+	if (c == '-' || c == '/')
+		return true;
+	return false;
+}
+
+bool Path::EscapeCmdLine(std::string* arg)
+{
+#ifdef _WIN32
+	if (arg->empty() || arg->find_first_of(" \t\"") == std::string::npos)
```

Comment:
I don't think this is a problem with the current usage of this functions, since no empty arguments are actually passed, but for the future it would be nice if this function didn't do nothing when the input string is empty.

The [code example from Microsoft that this is based on](https://learn.microsoft.com/en-gb/archive/blogs/twistylittlepassagesallalike/everyone-quotes-command-line-arguments-the-wrong-way#the-correct-solution) does the opposite, it checks that the string is not empty before returning (and uses && instead of ||), otherwise empty arguments will be dropped.